modules/core/src/main/scala/data/TransactionAccessMode.scala

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2018-2020 by Rob Norris
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package skunk.data
+
+/** Enumerated type of transaction access mode values. */
+sealed abstract class TransactionAccessMode extends Product with Serializable
+object TransactionAccessMode {
+
+  case object ReadOnly extends TransactionAccessMode
+
+  case object ReadWrite extends TransactionAccessMode
+
+  def toLiteral(accessMode: TransactionAccessMode): String =
```

Comment:
I think a `private[skunk] sql: String` member on `TransactionAccessMode` and `TransactionIsolationLevel` might be an easier way to do this.

Reply:
Looking much better, yes :)